dbms/src/AggregateFunctions/AggregateFunctionGroupArray.h

```diff
@@ -96,6 +235,17 @@ class GroupArrayNumericImpl final
         size_t size = value.size();
         writeVarUInt(size, buf);
         buf.write(reinterpret_cast<const char *>(value.data()), size * sizeof(value[0]));
+
+        if constexpr (Trait::sampler == Sampler::RNG)
+        {
+            DB::writeIntBinary<size_t>(this->data(place).total_values, buf);
+            std::ostringstream rng_stream;
```

Comment:
Looks very painful. Maybe rng is POD and we can just writePODBinary?

Reply:
unfortunately it's not POD

Reply:
Technically it's not POD but it should behave this way.